cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala

```diff
@@ -82,40 +91,75 @@ class OpenAIPrompt(override val uid: String) extends Transformer
     import com.microsoft.azure.synapse.ml.core.schema.DatasetExtensions._
 
     logTransform[DataFrame]({
       val df = dataset.toDF
 
-      val promptColName = df.withDerivativeCol("prompt")
-
-      val dfTemplated = df.withColumn(promptColName, Functions.template(getPromptTemplate))
-
-      val completion = openAICompletion.setPromptCol(promptColName)
-
-      // run completion
-      val results = completion
-        .transform(dfTemplated)
-        .withColumn(getOutputCol,
-          getParser.parse(F.element_at(F.col(completion.getOutputCol).getField("choices"), 1)
-            .getField("text")))
-        .drop(completion.getOutputCol)
-
-      if (getDropPrompt) {
-        results.drop(promptColName)
-      } else {
-        results
+      val completion = openAICompletion
+      val promptCol = Functions.template(getPromptTemplate)
+      val systemPrompt = "You are an AI Chatbot. Only respond with a completion."
+      val createMessagesUDF = udf((userMessage: String) => {
+        Seq(
+          OpenAIMessage("system", systemPrompt),
+          OpenAIMessage("user", userMessage)
+        )
+      })
+      completion match {
+        case chatCompletion: OpenAIChatCompletion =>
+          val messageColName = df.withDerivativeCol("messages")
+          val dfTemplated = df.withColumn(messageColName, createMessagesUDF(promptCol))
+          val completionNamed = chatCompletion.setMessagesCol(messageColName)
+
+          val results = completionNamed
+            .transform(dfTemplated)
+            .withColumn(getOutputCol,
+              getParser.parse(F.element_at(F.col(completionNamed.getOutputCol).getField("choices"), 1)
+                .getField("message").getField("content")))
+            .drop(completionNamed.getOutputCol)
+
+          if (getDropMessages) {
+            results.drop(messageColName)
+          } else {
+            results
+          }
+
+        case completion: OpenAICompletion =>
+          val promptColName = df.withDerivativeCol("prompt")
+          val dfTemplated = df.withColumn(promptColName, promptCol)
+          val completionNamed = completion.setPromptCol(promptColName)
+
+          // run completion
+          val results = completionNamed
+            .transform(dfTemplated)
+            .withColumn(getOutputCol,
+              getParser.parse(F.element_at(F.col(completionNamed.getOutputCol).getField("choices"), 1)
+                .getField("text")))
+            .drop(completionNamed.getOutputCol)
+
+          if (getDropPrompt) {
+            results.drop(promptColName)
+          } else {
+            results
+          }
       }
     }, dataset.columns.length)
   }
 
-  private def openAICompletion: OpenAICompletion = {
-    // apply template
-    val completion = new OpenAICompletion()
+  private def openAICompletion: OpenAIServicesBase = {
 
+    val completion: OpenAIServicesBase =
+    // use OpenAICompletion
+    if (getDeploymentName != "gpt-4") {
```

Comment:
Am doing this check to determine if gpt-4 is being used. I don't think this is right - afaik the deployment name is set by the user when creating their AOAI resource. In addition, we should probably enable OpenAIChatCompletions for models other than gpt-4, like gpt-4o. Let me know what the proper way to do this is